src/commands/runtime/activation/list.js

```diff
@@ -62,40 +62,52 @@ class ActivationList extends RuntimeBaseCommand {
           },
           Kind: {
             get: (row) => {
-              const { annotations } = row
-              if (!annotations || !annotations.length) return
-              return annotations.find((elem) => {
-                return (elem.key === 'kind')
-              }).value
+              if (row.duration !== undefined) {
+                const { annotations } = row
+                if (!annotations || !annotations.length) return
+                return annotations.find((elem) => {
+                  return (elem.key === 'kind')
+                }).value
+              } else {
+                // this is a trigger
+                return 'trigger'
+              }
             }
           },
           Start: {
             get: (row) => {
-              const { annotations } = row
-              if (!annotations || !annotations.length) return
-              const elem = annotations.find((elem) => {
-                return (elem.key === 'initTime')
-              })
-              return elem ? 'cold' : 'warm'
+              if (row.duration !== undefined) {
+                const { annotations } = row
+                if (!annotations || !annotations.length) return
+                const elem = annotations.find((elem) => {
+                  return (elem.key === 'initTime')
+                })
+                return elem ? 'cold' : 'warm'
+              } else {
+                return '--'
+              }
             }
           },
           Duration: {
-            get: row => `${row.duration}ms`
+            get: row => row.duration ? `${row.duration}ms` : '--'
           },
           Status: {
             get: (row) => {
               const statusStrings = ['success', 'application error', 'developer error', 'internal error']
-              return statusStrings[row.statusCode]
+              return statusStrings[row.statusCode || 0]
```

Comment:
I see trigger activations already do send a statusCode of `0`. So do we need this change ?

Reply:
I don't believe `statusCode` is present on a trigger list. From the [indexing function](https://github.com/apache/openwhisk/blob/4babe39fd2dbcc900ccedb5a5e9561d301361205/ansible/files/whisks_design_document_for_activations_db_filters_v2.1.1.json#L6)

```js
    var endtime = doc.end !== 0 ? doc.end : undefined;
    statusCode: (endtime !== undefined && doc.response !== undefined && doc.response.statusCode !== undefined) ? doc.response.statusCode : undefined
```

The `end` item for a trigger is undefined by definition so statusCode should be `undefined` as well.

Reply:
Note that it may be that other deployments provide the `statusCode` for activations indiscriminately. This change is therefore defensive and allows `aio` to work with other openwhisk deployments.